pex/testing.py

```diff
@@ -345,7 +363,7 @@ def ensure_python_distribution(version):
   pyenv = os.path.join(pyenv_root, 'bin', 'pyenv')
   pip = os.path.join(interpreter_location, 'bin', 'pip')
 
-  if not os.path.exists(pyenv):
+  if not (os.path.exists(pyenv) and pyenv_has_test_interpreters(pyenv_root)):
```

Comment:
Since the touch file is only touched post pyenv clone, you just need the 1 check. Given that, just inlining makes sense:
```python
if not os.path.exists(os.path.join(pyenv_root, INTERPRETER_SET_FINGERPRINT)):
  bootstrap_python_installer(pyenv_root)
```